File: library/src/main/java/com/github/dmitrikudrenko/countryphonecodechooser/ui/BorderTransformation.java
package com.github.dmitrikudrenko.countryphonecodechooser.ui;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.support.annotation.Nullable;
import android.support.annotation.RestrictTo;
import android.support.annotation.RestrictTo.Scope;
import com.squareup.picasso.Transformation;

@RestrictTo(Scope.LIBRARY)
class BorderTransformation implements Transformation {
    private final int color;
    private final int borderSize;

    BorderTransformation(int color, int borderSize) {
        this.color = color;
        this.borderSize = borderSize;
    }

    @Override
    public Bitmap transform(@Nullable Bitmap source) {
        if (source == null) return null;

        final int borderPadding = borderSize * 2;
        final int width = source.getWidth() + borderPadding;
        final int height = source.getHeight() + borderPadding;
        final Bitmap output = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
        final Canvas canvas = new Canvas(output);
        canvas.drawColor(color);
        canvas.drawBitmap(source, borderSize, borderSize, null);
        source.recycle();
        return output;
    }

    @Override
    public String key() {
        return "BorderTransformation";
    }
}

File: library/src/main/kotlin/com/github/dmitrikudrenko/countryphonecodechooser/utils/BorderTransformation.kt
package com.github.dmitrikudrenko.countryphonecodechooser.utils

import android.graphics.Bitmap
import android.graphics.Canvas
import android.support.annotation.RestrictTo
import android.support.annotation.RestrictTo.Scope
import com.squareup.picasso.Transformation

@RestrictTo(Scope.LIBRARY)
class BorderTransformation(private val color: Int, private val borderSize: Int) : Transformation {
    private val key = "BorderTransformation, color:$color,borderSize:$borderSize"

    override fun transform(source: Bitmap?): Bitmap? {
        if (source == null) return null

        val borderPadding = borderSize * 2
        val width = source.width + borderPadding
        val height = source.height + borderPadding
        val output = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
        val canvas = Canvas(output)
        canvas.drawColor(color)
        canvas.drawBitmap(source, borderSize.toFloat(), borderSize.toFloat(), null)
        source.recycle()
        return output
    }

    override fun key(): String {
        return key
    }
}
